plugins/pulp_docker/plugins/importers/upload.py

```diff
@@ -337,16 +337,14 @@ def process_main(self, item=None):
         :return:     None
         """
         if isinstance(item, models.Blob):
-            blob_src_path = os.path.join(self.get_working_dir(), item.digest.split(':')[1] + '.tar')
+            checksum_type, _, checksum = item.digest.rpartition(':')
+            if not checksum_type:
+                # Never assume. But oh well
+                checksum_type = "sha256"
+            blob_src_path = os.path.join(self.get_working_dir(), checksum + '.tar')
             blob_dest_path = os.path.join(self.get_working_dir(), item.digest)
-            with open(blob_src_path) as blob_src:
-                with contextlib.closing(gzip.open(blob_dest_path, 'w')) as blob_dest:
-                    # these can be big files, so we chunk them
-                    reader = functools.partial(blob_src.read, 4096)
-                    for chunk in iter(reader, ''):
-                        blob_dest.write(chunk)
-            # we don't need the tarfile anymore
-            os.remove(blob_src_path)
+            if blob_src_path != blob_dest_path:
```

Comment:
and you probably don't need this condition as you just define these 2 vars above and you can see that they would never be ==